scipy/special/_digamma.pxd

```diff
@@ -17,12 +17,6 @@ from ._trig cimport sinpi, cospi
 from ._cephes cimport zeta, psi
 from . cimport sf_error
 
-# Use the asymptotic series for z away from the negative real axis
-# with abs(z) > smallabsz.
-DEF smallabsz = 16
-# Use the reflection principle for z with z.real < 0 that are within
-# smallimag of the negative real axis.
-DEF smallimag = 6
 # Relative tolerance for series
 DEF tol = 2.220446092504131e-16
 # All of the following were computed with mpmath
```

Comment:
Just  checking I understand correctly. If one of these constants only appears in a single function, then we can just put it in that function as a local variable, and the compiler will be smart enough to optimize things because it can figure out the variable won't change. If something appears in multiple functions, if we make it a global variable, the compiler cannot tell it won't change at compile time unless we hint it with `const` but Cython doesn't offer that yet.

`posroot` and `posrootval` only appear in one function, `cdigamma`, but by symmetry it's kind of weird to have `negroot` and `negrootval` as compile time constants but not `posroot` and `posrootval`. The code is easier to understand if they're all in the same place.

Reply:
Yes, all correct. We're anticipating `cdef const` to appear, and that's a nicer / more consistent change that moving `posroot` and `posrootval`.